tools/lib/tests/test_caching.py

```diff
@@ -59,6 +62,31 @@ def test_large_file(self):
     self.compare_loads(large_file_url, length - 100, 100)
     self.compare_loads(large_file_url)
 
+  @parameterized.expand([(True, ), (False, )])
+  def test_recover_from_missing_file(self, cache_enabled):
+    os.environ["FILEREADER_CACHE"] = "1" if cache_enabled else "0"
+
+    file_url = "http://localhost:5001/test.png"
+
+    file_exists = False
+
+    def get_length_online_mock(self):
+      if file_exists:
+        return 4
+      return -1
+
+    patch_length = mock.patch.object(URLFile, "get_length_online", get_length_online_mock)
+    patch_length.start()
+    try:
+      length = URLFile(file_url).get_length()
+      self.assertEqual(length, -1)
```

Comment:
If you run this test twice in a row with a clean download cache folder, it will fail the second time since it's not clearing the cache.

Reply:
are you running it directly? with pytest, it should use the openpilot_prefix and make a temp directory

Reply:
Oh, pytest works, but running the file and using unittest does not.

Reply:
hmm, we probably don't want to clear the full download cache if they run it directly (so they keep their local cached files), but maybe clear just that file